feat: card component

theocerutti opened the pull request:
**Component**: Card

**Description**:

**Design**:

![image](https://user-images.githubusercontent.com/44285344/118259960-10904200-b4b2-11eb-9252-b2e25a50711b.png)

**Breaking Changes**:

- [NO] Breaking Changes

Closes: #164

bert-e commented:
# Hello theocerutti,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/core-ui/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Conflict




There is a conflict between your branch `feature/card-component` and the
destination branch `development/1.0`.

Please resolve the conflict on **the feature branch** (`feature/card-component`).

```bash
 $ git fetch
 $ git checkout origin/feature/card-component
 $ git merge origin/development/1.0
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:feature/card-component
```

JBWatenbergScality commented on src/lib/components/card/Card.component.js:
```diff
@@ -0,0 +1,105 @@
+  border-collapse: collapse;
+  box-sizing: border-box;
+
+  &:hover {
+    ${props => {
+      const borderColor = getTheme(props)[props.borderColor]
+      return css`
+        border: 2px solid ${borderColor};
+        border-radius: 8px;
+      `;
+    }}
+  }
+  
+  &:hover > div {
+    background-image: linear-gradient(rgba(${props => hex2RGB(getTheme(props)[props.borderColor]).join(',')}, 0.05) 0 0);
+  }
```
I think it is missing an active and focus state as well as keyboard navigation. You can get inspired from this gist https://gist.github.com/JBWatenbergScality/b5975905defdcab9c4300da2f7e19cc5

ChengYanJin reviewed (commented):
You may want to address the conflicts, thanks

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @nicolas2bert

* @carlito-scality

* @monpote

* @chengyanjin

* @jbwatenbergscality

* @rachedbenmustapha

JBWatenbergScality commented:
Thanks for the changes. We are waiting for https://github.com/scality/core-ui/pull/295 to land before merging this with the new theme.

bert-e commented:
# Conflict




There is a conflict between your branch `feature/card-component` and the
destination branch `development/1.0`.

Please resolve the conflict on **the feature branch** (`feature/card-component`).

```bash
 $ git fetch
 $ git checkout origin/feature/card-component
 $ git merge origin/development/1.0
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:feature/card-component
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @nicolas2bert

* @carlito-scality

* @monpote

* @chengyanjin

* @jbwatenbergscality

* @rachedbenmustapha

JBWatenbergScality commented:
Hi @theocerutti ,
We changed a bit the design of the card. Now the hover/focused border should be an outline and use the `highlight`  theme color. Could you please sync the card implementation with those changes ?

Thanks,
Jean-Baptiste

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @monpote

* @rachedbenmustapha

* @carlito-scality

* @jbwatenbergscality

* @nicolas2bert

* @chengyanjin

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @chengyanjin

* @rachedbenmustapha

* @carlito-scality

* @monpote

* @jbwatenbergscality

* @nicolas2bert

theocerutti commented:
@JBWatenbergScality 

> Hi @theocerutti ,
> We changed a bit the design of the card. Now the hover/focused border should be an outline and use the `highlight` theme color. Could you please sync the card implementation with those changes ?
> 
> Thanks,
> Jean-Baptiste

Done (i updated pictures in the PR description).

I also modified the default color value to match darkRebrand theme:
```js
  headerBackgroundColor = 'backgroundLevel1',
  bodyBackgroundColor = 'backgroundLevel2',
  borderColor = 'highlight', // hover and border
```
is it okay ?

JBWatenbergScality commented on src/lib/components/card/Card.component.js:
```diff
@@ -0,0 +1,134 @@
+        box-shadow: 0 0 0 2px ${getTheme(props).highlight};
+      }
+  
+      &:focus {
+        outline: 0;
+        box-shadow: 0 0 0 1px ${getTheme(props)[props.status || 'highlight']};
+      }
```
@Cuervino what do you think of this `focus` state for the cards ?

Cuervino replied:
IMO, the focus state like that is too close to the hover state.
There should be a "additive" logic.
I think that an outline 2 px out of the card with the color buttonSecondary should be good.
<img width="166" alt="navigation-status=Focus, Health=healthy" src="https://user-images.githubusercontent.com/28654802/118679684-5de81880-b7fe-11eb-8556-be70294198c9.png">

JBWatenbergScality commented on src/lib/components/card/Card.component.js:
```diff
@@ -0,0 +1,134 @@
+  body?: Node,
+  headerBackgroundColor?: string,
+  bodyBackgroundColor?: string,
+  status?: 'healthy' | 'warning' | 'critical',
```
An healthy card should use `backgroundLevel4` for the header background (current default if no status is provided)

Cuervino replied:
Correct, the logic is that we'll not focus too much on the Green Healthy part, that could distract the user, but we'll highlight any non-healthy state with Warning and Critical colors.

bert-e commented:
# Conflict




There is a conflict between your branch `feature/card-component` and the
destination branch `development/1.0`.

Please resolve the conflict on **the feature branch** (`feature/card-component`).

```bash
 $ git fetch
 $ git checkout origin/feature/card-component
 $ git merge origin/development/1.0
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:feature/card-component
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @chengyanjin

* @rachedbenmustapha

* @carlito-scality

* @monpote

* @jbwatenbergscality

* @nicolas2bert

bert-e commented:
# Conflict




There is a conflict between your branch `feature/card-component` and the
destination branch `development/1.0`.

Please resolve the conflict on **the feature branch** (`feature/card-component`).

```bash
 $ git fetch
 $ git checkout origin/feature/card-component
 $ git merge origin/development/1.0
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:feature/card-component
```

theocerutti committed:
Card: add card component

theocerutti committed:
Card: update snapshots

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @chengyanjin

* @rachedbenmustapha

* @carlito-scality

* @monpote

* @jbwatenbergscality

* @nicolas2bert

theocerutti commented:
@bert-e approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/1.0`




There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/1.0`




Please check the status of the associated issue None.

Goodbye theocerutti.
